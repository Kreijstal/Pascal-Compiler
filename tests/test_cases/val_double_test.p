program val_double_test;
var
  s: string;
  value: Double;
  code: Integer;
  number: Integer;
begin
  s := '3.1415';
  value := 0.0;
  code := -1;
  number := -1;
  Val(s, value, code);
  if (code = 0) and (value > 3.0) then
    WriteLn('Value: ', value:0:4, ' Code: ', code);

  s := '42';
  Val(s, number, code);
  WriteLn('Integer: ', number, ' Code: ', code);

  s := '7Z';
  number := -5;
  Val(s, number, code);
  WriteLn('Invalid: ', number, ' Code: ', code);
end.
