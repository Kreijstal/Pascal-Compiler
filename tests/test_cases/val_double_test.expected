Value: 3.1415 Code: 0
Integer: 42 Code: 0
Invalid: -5 Code: 2
